stdlib/jdbc/src/test/java/org/ballerinax/jdbc/actions/SelectTest.java

```diff
@@ -0,0 +1,124 @@
+        Assert.assertEquals(((BDecimal) returns[6]).decimalValue(), new BigDecimal("1234.567"));
+        Assert.assertEquals(((BFloat) returns[7]).floatValue(), 1234.567, 0.001);
+        Assert.assertEquals(((BFloat) returns[8]).floatValue(), 1234.567, 0.001);
+    }
+
+    @Test(groups = SELECT_TEST)
+    public void testSelectNumericDataWithDBError() {
+        BValue[] returns = BRunUtil.invoke(result, "testSelectNumericDataWithDBError");
+        Assert.assertTrue(((BBoolean) returns[0]).booleanValue());
+        Assert.assertTrue(((BBoolean) returns[1]).booleanValue());
+        Assert.assertTrue(((BBoolean) returns[2]).booleanValue());
+        Assert.assertFalse(((BBoolean) returns[3]).booleanValue());
+        Assert.assertTrue(returns[4].stringValue().contains("Failed to execute select query:"));
+        Assert.assertEquals(returns[5].stringValue(), "42501");
+        Assert.assertEquals(returns[6].stringValue(),"{ballerinax/java.jdbc}DatabaseError");
+        Assert.assertEquals(((BInteger) returns[7]).intValue(), -5501);
+    }
+
+    @Test(groups = SELECT_TEST)
+    public void testSelectNumericDataWithApplicationError() {
+        BValue[] returns = BRunUtil.invoke(result, "testSelectNumericDataWithApplicationError");
+        Assert.assertTrue(((BBoolean) returns[0]).booleanValue());
+        Assert.assertTrue(((BBoolean) returns[1]).booleanValue());
+        Assert.assertFalse(((BBoolean) returns[2]).booleanValue());
+        Assert.assertTrue(((BBoolean) returns[3]).booleanValue());
+        Assert.assertEquals(returns[4].stringValue(), "{ballerinax/java.jdbc}ApplicationError");
+        Assert.assertTrue(
+                returns[5].stringValue().contains("Unsupported array type specified as a parameter at index 0"));
+    }
+
+    @Test(groups = SELECT_TEST)
+    public void testArrayOfQueryParameters() {
+        BValue[] returns = BRunUtil.invoke(result, "testArrayOfQueryParameters");
+        Assert.assertEquals(returns[0].stringValue(), "Peter");
+    }
+
+    @Test(groups = SELECT_TEST)
+    public void testBoolArrayOfQueryParameters() {
+        BValue[] returns = BRunUtil.invoke(result, "testBoolArrayOfQueryParameters");
+        Assert.assertEquals(returns[0].stringValue(), "Peter");
+    }
+
+    @Test(groups = SELECT_TEST)
+    public void testBlobArrayOfQueryParameter() {
+        BValue[] returns = BRunUtil.invoke(result, "testBlobArrayOfQueryParameter");
+        Assert.assertEquals(returns[0].stringValue(), "Peter");
+    }
+
+    @Test(dependsOnGroups = SELECT_TEST)
+    public void testCloseConnectionPool() {
+        BValue[] returns = BRunUtil.invoke(result, "testCloseConnectionPool");
+        BInteger retValue = (BInteger) returns[0];
+        Assert.assertEquals(retValue.intValue(), 1);
+    }
+
+    @AfterSuite
+    public void cleanup() {
+        if (testDatabase != null) {
+            testDatabase.stop();
+        }
+    }
```

Comment:
I suppose the reason for not having `select` tests for other types is because they are included in the UpdateTest?

Reply:
No. We need to gradually improve that class to improve coverage, remove unnecessary duplicate scenarios, cover all data types etc. Updated the PR description with more details